In [2]:
import numpy as np

a=np.array([1,2,3,4])
print(a)

[1 2 3 4]


In [5]:
import time
a=np.random.rand(1000000)
b=np.random.rand(1000000)

In [7]:
a.shape

(1000000,)

In [8]:
a

array([0.48593652, 0.18452281, 0.43369753, ..., 0.86155836, 0.85834857,
       0.17813198])

### time.time() represents the current time in seconds since the Unix epoch (January 1, 1970, 00:00:00 UTC)

In [9]:
time.time()

1721396507.0078537

import time

timestamp = 1721396507.0078537
readable_time = time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(timestamp))
print(readable_time)

In [24]:
a=np.random.rand(1000000)
b=np.random.rand(1000000)

In [29]:
import time

tic=time.time()
c=np.dot(a,b)
toc=time.time()

print(c)
print("Vectorize version: "+str(1000*(toc-tic))+'ms')

250318.81711926535
Vectorize version: 1.0030269622802734ms


In [30]:
c=0
tic=time.time()
for i in range(0,1000000):
    c += a[i]*b[i]
toc=time.time()

print(c)
print("For loop version: "+str(1000*(toc-tic))+'ms')

250318.81711927583
For loop version: 373.3501434326172ms


In [31]:
373.3501434326172/1.0030269622802734

372.2234371285952

## vertorized code run faster